client/lib/products-values/index.js

```diff
@@ -359,6 +364,25 @@ export function getDomain( product ) {
 	return product.meta;
 }
 
+export function getJetpackProductDisplayName( product ) {
+	product = formatProduct( product );
+	assertValidProduct( product );
+
+	return JETPACK_PRODUCT_DISPLAY_NAMES?.[ product?.productSlug ];
```

Comment:
Wont the previous `formatProduct` and `assertValidProduct` ensure that we have a valid `product` with a valid `productSlug`, making this `product?.` optional chaining unnecessary?